pkg/distributor/distributor.go

```diff
@@ -1208,6 +1208,9 @@ func (d *Distributor) prePushValidationMiddleware(next PushFunc) PushFunc {
 		var firstPartialErr error
 		var removeIndexes []int
 		totalSamples, totalExemplars := 0, 0
+		// Cap how many metrics with dedupled samples to emit trace events for.
+		const maxTrace = 10
```

Comment:
```suggestion
		const maxMetricsWithDeduplicatedSamplesToTrace = 10
```

Then you don't need a comment.